doc/02_architecture.md

```diff
@@ -0,0 +1,39 @@
+# Software Architecture {#architecture} #
+As mentioned in [Introduction](@ref mainpage), the `asteroidsyncservice` is just one component of a synchronization client.  This section describes a whole synchronization and shows how this software fits into the overall picture.  This project is intended to be used by a [QML](https://doc.qt.io/qt-5/qmlapplications.html) application without the application needing to know about the details of this software.  The `asteroidsyncservice` contains two major components: a QML plugin and a daemon that it interacts with.  
+
+\startuml
+[QML app]
+package "asteroidsyncservice" {
+    [QML plugin] - qml
+    [QML app] --> qml
+    [daemon] - dbus
+    [QML plugin] --> dbus
+}
+[system] 
+dbus --> [system]
+\enduml
+
+When an external QML application wants to use this software it uses the line `import org.asteroid.syncservice 1.0` and then can instantiate a regular QML object:
+
+```
+ServiceController {
+    id: serviceController
+    Component.onCompleted: serviceController.startService()
+    Component.onDestruction: serviceController.stopService()
+}
+```
+
+In the SyncServicePlugin::registerTypes function, the plugin exports three types: Watch, Watches and ServiceControl which is exported to QML as `ServiceController` as shown above.  Once the `ServiceController` is started, the Watches collection of Watch objects may be used to interact with a selected watch.
+
+Consult the details of those classes to read about the signals and properties that are available to a QML application.
+
+## Platform support ##
+The QML plugin provides two different ways to automatically start the daemon.  One is via [D-Bus](@ref dbus), which is used by the SailfishOS platform and the Linux desktop platform, and the other is via `upstart` which is used by the Ubuntu Touch platform.
```

Comment:
Iiuc all these "@ref" are supposed to be accompanied by glossary entries ? Are these entries you forgot to document or do I misunderstand @ref ?

Reply:
Yes, each of the `@ref` entries refers to some other anchor in the documentation.  In some cases it refers to a glossary entry and in others, such as `@ref dbus` it refers to a page title.